lib/utils/mailup_nodemailer.ts

```diff
@@ -0,0 +1,144 @@
+/**
+ * Nodemailer transport for MailUp transactional APIs
+ * 
+ * see http://help.mailup.com/display/mailupapi/Transactional+Emails+using+APIs
+ * and https://nodemailer.com/plugins/create/#transports
+ * 
+ */
+import * as nodemailer from "nodemailer";
+
+import { Address } from "nodemailer/lib/addressparser";
+
+import * as mailup from "./mailup";
+
+import * as winston from "winston";
+
+import { isRight } from "fp-ts/lib/Either";
+
+const TRANSPORT_NAME = "MailUp";
+const TRANSPORT_VERSION = "0.1";
+
+interface IMailUpTransportOptions {
+  readonly creds: mailup.ISmtpAuthInfo;
+}
+
+interface IAddresses {
+  readonly from?: ReadonlyArray<Address>;
+  readonly sender?: ReadonlyArray<Address>;
+  readonly "reply-to"?: ReadonlyArray<Address>;
+  readonly to?: ReadonlyArray<Address>;
+  readonly cc?: ReadonlyArray<Address>;
+  readonly bcc?: ReadonlyArray<Address>;
+}
+
+/*
+ * Usage:
+ *
+ * const transporter = nodemailer.createTransport(
+ *   MailUpTransport({
+ *     creds: {
+ *       Username: <SMPT+Username>,
+ *       Secret: <SMPT+Password>
+ *     }
+ *   })
+ * );
+ * 
+ * transporter
+ *   .sendMail({
+ *     from:      "foobar@example.com",
+ *     to:        "deadbeef@example.com",
+ *     replyTo:   "foobar-reply@example.com",
+ *     subject:   "lorem ipsum",
+ *     text:      "lorem ipsum",
+ *     html:      "<b>lorem ipsum</b>"
+ *   })
+ *   .then(res => console.log(JSON.stringify(res)))
+ *   .catch(err => console.error(JSON.stringify(err)));
+ */
+export function MailUpTransport(
+  options: IMailUpTransportOptions
+): nodemailer.Transport {
+  return {
+    name: TRANSPORT_NAME,
+
+    version: TRANSPORT_VERSION,
+
+    // tslint is wrong here, this *has* a typedef
+    // tslint:disable-next-line:typedef
+    async send(mail, callback) {
```

Comment:
perhaps tslint is happier with the form `send: async (mail, cb) => {}`

Reply:
yes =)